src/Model/Subscription.php

```diff
@@ -236,6 +272,7 @@ public function getAddOns(): array
     public function setAddOns(Modifiers $addOns): self
     {
         $this->addOns = $addOns;
+        $this->previousAddOns = clone $addOns;
```

Comment:
What about `addDiscount` and `addAddon`, `removeAddOn`, etc....

This seems like that could bite us if we allow things to get out of sync when adding or removing individual modifiers right?

Reply:
Maybe I don't understand the use case? Will we really want to clone and track the last set of previous modifiers every time a change occurs? Or are we just interested in the "initial" set of modifiers?

Reply:
This was more to cover what was originally coming back from Braintree. There was some logic later on that tried to determine if we should add a `Modifier` or update an existing one... and if it was non-default and already on the `Subscription` it would fail when we'd make the request to Braintree - since we thought it needed to be added, but it already existed and needed to be updated.

Being able to compare it to the Braintree Subscription's `Modifiers` when fetched gives us the chance to catch that case.

To be honest, I struggled with this part a good bit and couldn't come up with a better way to resolve the issue mentioned above. I'm game to pair with your for a little bit if we want to see if we can come up with a different solution that doesn't include storing the defaults from the subscription in BT.

Reply:
I guess I just want to understand what `storing the defaults` means. We set the previous values when we call `setDiscounts` or `setAddOns`, but we don't track the previous values when adding/removing individual ones.

Is this intentional? If so I think we have competing concepts and maybe we should reason about it in a *slightly* different way.